app/services/memberships/revoke_service.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+module Memberships
+  class RevokeService < BaseService
+    def revoke(id)
+      membership = Membership.find_by(id: id)
+      return result.fail!(code: 'not_found') unless membership
```

Comment:
What about returning an explicit error code?

```suggestion
      return result.fail!(code: 'membership_not_found') unless membership
```